modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/extras/MatchOnlyTextFieldMapper.java

```diff
@@ -547,8 +573,12 @@ protected void parseCreateField(DocumentParserContext context) throws IOExceptio
         context.addToFieldNames(fieldType().name());
 
         if (storeSource) {
-            final var bytesRef = new BytesRef(utfBytes.bytes(), utfBytes.offset(), utfBytes.length());
-            context.doc().add(new StoredField(fieldType().storedFieldNameForSyntheticSource(), bytesRef));
+            if (storedFieldInBinaryFormat) {
+                final var bytesRef = new BytesRef(utfBytes.bytes(), utfBytes.offset(), utfBytes.length());
+                context.doc().add(new StoredField(fieldType().storedFieldNameForSyntheticSource(), bytesRef));
+            } else {
+                context.doc().add(new StoredField(fieldType().storedFieldNameForSyntheticSource(), value.string()));
+            }
```

Comment:
Not 100% sure about this. Maybe we want to continue writing in binary format even for older indices? Since we already updated the mapper to handle mixed string and byteref values, we may as well take advantage of the throughput benefits?

Reply:
I think this way the logic is simpler? So maybe keep it like this? Indices will rollover at some point and then binary format will be used.